library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/scalardd/GetRDDOfElementsHandlerTest.java

```diff
@@ -377,4 +377,22 @@ public void checkGetCorrectElementsInRDDForEdgeId() throws OperationException, I
         sparkContext.stop();
     }
 
+    @Test(expected = java.lang.IllegalArgumentException.class)
```

Comment:
Although it is neater to use the expected exception annotation it means the unit test will pass if an illegal argument exception is thrown from anywhere in the test. We only want the test to pass if the exception is thrown when graph.execute is called. For that reason I tend to use try/catch to ensure I am definitely testing the exception is thrown in the right place. E.g:

try {
    graph1.execute(rddQuery, user);
    fail("Exception expected");
} catch(final IllegalArgumentException e) {
    assertNotNull(e.getMessage());
}